### ISSUE TITLE
Unable to reach auc results showed in paper

### ISSUE THREAD

**Issue by daviduarte** [tianyu0207/RTFM#18]
Hello! First, congratulaton for the excelent paper. I'm trying training the network from scratch using UCF Crimes dataset, starting with the I3D features available in this repo (I do not extracted the I3D features by my self). However, setting the same hyperparameters as paper (0,001 initial LR, 0.0005 weight decay, 64 batch size, 50 epochs and same dataset division) I'm unable to reach the 0.84 of AUC, even after 150 epochs. The greater AUC achieved was near 0.63 (curiosly in the end of training). 

Please, look the charts bellow.

![newplot](https://user-images.githubusercontent.com/6354107/119134860-0cb17200-ba14-11eb-9f56-1dd0b2bd813e.png)
![newplot (1)](https://user-images.githubusercontent.com/6354107/119134865-0d4a0880-ba14-11eb-912c-75b79c02bb6e.png)
![newplot (2)](https://user-images.githubusercontent.com/6354107/119134868-0d4a0880-ba14-11eb-8039-36539ddd43ca.png)
![newplot (3)](https://user-images.githubusercontent.com/6354107/119134871-0de29f00-ba14-11eb-857f-9691b02fe277.png)
![newplot (4)](https://user-images.githubusercontent.com/6354107/119134872-0de29f00-ba14-11eb-96bc-1821560680ca.png)
![newplot (5)](https://user-images.githubusercontent.com/6354107/119134874-0e7b3580-ba14-11eb-9141-c1262691eb87.png)
![newplot (6)](https://user-images.githubusercontent.com/6354107/119134875-0f13cc00-ba14-11eb-9ade-8f58f2f57676.png)

I'm ploting the test AUC every 5 epochs, and the training loss is plotted every epoch. Notice that, in the 5º epoch, the test AUC reaches the greater value (near 0.66), then suddenly the measure drops down, while the training loss drops down during all training. In my opinion, these charts suggest overfitting (error in test set increasing and error in training set decreasing) >maybe< due high Learning Rate, but I trained again with LR 0.0001 (one magnitude order bellow than previous training) and the behavior was similar.

Maybe the hard coded dataset division in the dataset.py **self.list = self.list[63:]** and **self.list = self.list[:63]** was setted for the Shangay not for UCF?

Can you kindle suggest me some modification?

**Comment by tianyu0207** [comment]
> Hello! First, congratulaton for the excelent paper. I'm trying training the network from scratch using UCF Crimes dataset, starting with the I3D features available in this repo (I do not extracted the I3D features by my self). However, setting the same hyperparameters as paper (0,001 initial LR, 0.0005 weight decay, 64 batch size, 50 epochs and same dataset division) I'm unable to reach the 0.84 of AUC, even after 150 epochs. The greater AUC achieved was near 0.63 (curiosly in the end of training).
> 
> Please, look the charts bellow.
> 
> ![newplot](https://user-images.githubusercontent.com/6354107/119134860-0cb17200-ba14-11eb-9f56-1dd0b2bd813e.png)
> ![newplot (1)](https://user-images.githubusercontent.com/6354107/119134865-0d4a0880-ba14-11eb-912c-75b79c02bb6e.png)
> ![newplot (2)](https://user-images.githubusercontent.com/6354107/119134868-0d4a0880-ba14-11eb-8039-36539ddd43ca.png)
> ![newplot (3)](https://user-images.githubusercontent.com/6354107/119134871-0de29f00-ba14-11eb-857f-9691b02fe277.png)
> ![newplot (4)](https://user-images.githubusercontent.com/6354107/119134872-0de29f00-ba14-11eb-96bc-1821560680ca.png)
> ![newplot (5)](https://user-images.githubusercontent.com/6354107/119134874-0e7b3580-ba14-11eb-9141-c1262691eb87.png)
> ![newplot (6)](https://user-images.githubusercontent.com/6354107/119134875-0f13cc00-ba14-11eb-9ade-8f58f2f57676.png)
> 
> I'm ploting the test AUC every 5 epochs, and the training loss is plotted every epoch. Notice that, in the 5º epoch, the test AUC reaches the greater value (near 0.66), then suddenly the measure drops down, while the training loss drops down during all training. In my opinion, these charts suggest overfitting (error in test set increasing and error in training set decreasing) >maybe< due high Learning Rate, but I trained again with LR 0.0001 (one magnitude order bellow than previous training) and the behavior was similar.
> 
> Maybe the hard coded dataset division in the dataset.py **self.list = self.list[63:]** and **self.list = self.list[:63]** was setted for the Shangay not for UCF?
> 
> Can you kindle suggest me some modification?

Hi Yes it was the hardcoded code self.list = self.list[63:]** and **self.list = self.list[:63] is for Shanghai not for UCF. You can have a look of the UCF training list in github. I think the correct one for UCF should be self.list = self.list[810:]** and **self.list = self.list[:810]

**Comment by daviduarte** [comment]
I got much better results. Thank You!
